tests/test_unit.py

```diff
@@ -432,37 +383,102 @@ def test_extract_docstring_info_no_summary(self):
+                    },
+                },
+                "returns": [{
+                    "var_type": "str",
+                    "description": "simple description for return value.",
+                }],
+                "exceptions": [{
+                    "var_type": "AttributeError",
+                    "description": "if `condition x`.",
+                }],
+            },
+
+        ],
+    ]
+    @parameterized.expand(test_entries)
+    def test_extracts_docstring_info(
+        self,
+        summary,
+        expected_top_summary,
+        expected_summary_info
+    ):
+        summary_info = {
+            "variables": {},
+            "returns": [],
+            "exceptions": []
         }
 
-        top_summary_got = extension._extract_docstring_info(summary_info_got, summary, "")
-        self.assertEqual(top_summary_got, "")
-        self.assertDictEqual(summary_info_got, self.summary_info1_want)
+        top_summary = extension._extract_docstring_info(
+            summary_info,
+            summary,
+            "",
+        )
+
+        self.assertEqual(top_summary, expected_top_summary)
+        self.assertDictEqual(summary_info, expected_summary_info)
 
 
-    def test_find_package_group(self):
-        package_group_want = "google.cloud.spanner_v1beta2"
+    test_entries = [
+        [
+            """
+Description of docstring which should fail.
+
+:returns:param:
+            """,
+            ValueError,
+            "Error string",
+        ],
+        [
+            """
+Description of malformed docstring.
+
+Raises:
+    Error that should fail: if condition `x`.
+            """,
+            KeyError,
+            "Malformed docstring",
+        ],
+    ]
+    @parameterized.expand(test_entries)
+    def test_raises_error_extracting_malformed_docstring(
+        self,
+        summary,
+        error_type,
+        error_string,
+    ):
+        with self.assertRaises(error_type):
+            extension._extract_docstring_info({}, summary, error_string)
+
+
+    def test_finds_package_group(self):
         uid = "google.cloud.spanner_v1beta2.services.admin_database_v1.types"
 
-        package_group_got = extension.find_package_group(uid)
-        self.assertEqual(package_group_got, package_group_want)
+        package_group = extension.find_package_group(uid)
+        self.assertEqual(package_group, "google.cloud.spanner_v1beta2")
 
 
-    def test_pretty_package_name(self):
-        package_name_want = "Spanner V1beta2"
+    def test_finds_pretty_package_name(self):
         package_group = "google.cloud.spanner_v1beta2"
 
-        package_name_got = extension.pretty_package_name(package_group)
-        self.assertEqual(package_name_got, package_name_want)
+        package_name = extension.pretty_package_name(package_group)
+        self.assertEqual(package_name, "Spanner V1beta2")
 
 
-    def test_group_by_package(self):
-        toc_yaml_want = [
+    def test_groups_by_package(self):
```

Comment:
Same as above, this test might also benefit from having the data being pulled out of the test logic.

Reply:
Done! I forgot to clean up this variable below as well.